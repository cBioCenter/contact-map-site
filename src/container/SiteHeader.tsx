import * as React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Header, Menu } from 'semantic-ui-react';

import { IContactMapSiteState } from '~contact-map-site~/reducer';

export interface ISiteHeaderProps {
  pathname: string;
}

export interface ISiteHeaderState {
  currentPageName: null | string;
}

export class UnconnectedSiteHeader extends React.Component<ISiteHeaderProps, ISiteHeaderState> {
  constructor(props: ISiteHeaderProps) {
    super(props);
    this.state = {
      currentPageName: null,
    };
  }

  public componentDidMount() {
    this.setState({
      currentPageName: this.props.pathname,
    });
  }

  public componentDidUpdate(prevProps: ISiteHeaderProps) {
    if (this.props.pathname && this.props.pathname !== this.state.currentPageName) {
      this.setState({
        currentPageName: this.props.pathname,
      });
    }
  }

  public render() {
    return (
      <Header>
        <Menu secondary={true} borderless={true} fluid={true} style={{ maxHeight: '40px', padding: '20px 0 0 0' }}>
          <Menu.Item fitted={'vertically'} position={'left'}>
            <Link to={'/'}>
              <img
                alt={'hca-dynamics-icon'}
                src={'assets/icons/bio-blocks-icon-2x.png'}
                style={{ height: '32px', width: '32px' }}
              />
              <span style={{ color: 'black', fontSize: '32px', fontWeight: 'bold' }}>
                ContactMap.org: 2D and 3D Visualization
              </span>
            </Link>
          </Menu.Item>
        </Menu>
        {this.renderNavMenu()}
      </Header>
    );
  }

  protected renderNavMenu = () => {
    const { currentPageName } = this.state;

    return (
      <Menu defaultActiveIndex={-1} secondary={true}>
        <Menu.Item key={'home'}>
          <Link
            to={'/'}
            style={{ color: 'black', fontSize: '18px', fontWeight: currentPageName === '/' ? 'bold' : 'normal' }}
          >
            Home
          </Link>
        </Menu.Item>
        <Menu.Item key={'manuscript'}>
          <Link
            to={'/manuscript'}
            style={{
              color: 'black',
              fontSize: '18px',
              fontWeight: currentPageName === '/manuscript' ? 'bold' : 'normal',
            }}
          >
            Manuscripts
          </Link>
        </Menu.Item>
      </Menu>
    );
  };
}

const mapStateToProps = (state: IContactMapSiteState) => ({
  pathname: state.router.location.pathname,
});

export const SiteHeader = connect(mapStateToProps)(UnconnectedSiteHeader);
