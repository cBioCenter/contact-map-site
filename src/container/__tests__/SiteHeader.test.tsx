import { shallow } from 'enzyme';
import * as React from 'react';

import { UnconnectedSiteHeader } from '~contact-map-site~/container';

describe('SiteHeader', () => {
  it('Should match existing snapshot when no props are provided.', () => {
    const wrapper = shallow(<UnconnectedSiteHeader pathname={'/'} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should update the menu buttons when changing pages.', () => {
    const wrapper = shallow(<UnconnectedSiteHeader pathname={''} />);
    wrapper.setProps({
      pathname: '/manuscript',
    });
    expect(wrapper).toMatchSnapshot();
  });
});
