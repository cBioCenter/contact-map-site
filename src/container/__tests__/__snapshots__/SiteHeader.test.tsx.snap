// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteHeader Should match existing snapshot when no props are provided. 1`] = `
<Header>
  <Menu
    borderless={true}
    fluid={true}
    secondary={true}
    style={
      Object {
        "maxHeight": "40px",
        "padding": "20px 0 0 0",
      }
    }
  >
    <MenuItem
      fitted="vertically"
      position="left"
    >
      <Link
        replace={false}
        to="/"
      >
        <img
          alt="hca-dynamics-icon"
          src="assets/icons/bio-blocks-icon-2x.png"
          style={
            Object {
              "height": "32px",
              "width": "32px",
            }
          }
        />
        <span
          style={
            Object {
              "color": "black",
              "fontSize": "32px",
              "fontWeight": "bold",
            }
          }
        >
          ContactMap.org: 2D and 3D Protein Visualization
        </span>
      </Link>
    </MenuItem>
  </Menu>
  <Menu
    defaultActiveIndex={-1}
    secondary={true}
  >
    <MenuItem
      key="home"
    >
      <Link
        replace={false}
        style={
          Object {
            "color": "black",
            "fontSize": "18px",
            "fontWeight": "bold",
          }
        }
        to="/"
      >
        Home
      </Link>
    </MenuItem>
    <MenuItem
      key="manuscript"
    >
      <Link
        replace={false}
        style={
          Object {
            "color": "black",
            "fontSize": "18px",
            "fontWeight": "normal",
          }
        }
        to="/manuscript"
      >
        Manuscripts
      </Link>
    </MenuItem>
  </Menu>
</Header>
`;

exports[`SiteHeader Should update the menu buttons when changing pages. 1`] = `
<Header>
  <Menu
    borderless={true}
    fluid={true}
    secondary={true}
    style={
      Object {
        "maxHeight": "40px",
        "padding": "20px 0 0 0",
      }
    }
  >
    <MenuItem
      fitted="vertically"
      position="left"
    >
      <Link
        replace={false}
        to="/"
      >
        <img
          alt="hca-dynamics-icon"
          src="assets/icons/bio-blocks-icon-2x.png"
          style={
            Object {
              "height": "32px",
              "width": "32px",
            }
          }
        />
        <span
          style={
            Object {
              "color": "black",
              "fontSize": "32px",
              "fontWeight": "bold",
            }
          }
        >
          ContactMap.org: 2D and 3D Protein Visualization
        </span>
      </Link>
    </MenuItem>
  </Menu>
  <Menu
    defaultActiveIndex={-1}
    secondary={true}
  >
    <MenuItem
      key="home"
    >
      <Link
        replace={false}
        style={
          Object {
            "color": "black",
            "fontSize": "18px",
            "fontWeight": "normal",
          }
        }
        to="/"
      >
        Home
      </Link>
    </MenuItem>
    <MenuItem
      key="manuscript"
    >
      <Link
        replace={false}
        style={
          Object {
            "color": "black",
            "fontSize": "18px",
            "fontWeight": "bold",
          }
        }
        to="/manuscript"
      >
        Manuscripts
      </Link>
    </MenuItem>
  </Menu>
</Header>
`;
